client/src/components/SortDirection/SortDirection.tsx

```diff
@@ -38,9 +45,24 @@ export const SortDirection = ({
   </button>
 );
 
-export const SortDirections = ({ asc, desc }: SortDirectionsProps) => (
+export const SortDirections = ({
+  asc,
+  desc,
+  sortFunction,
+  col,
+}: SortDirectionsProps) => (
```

Comment:
IMO, `SortDirections` shouldn't need to know about `col`. `col` is a concept leaking down from the current use cases. It could be possible that this is used in a context where columns aren't meaningful (i.e. maybe sorting a list rather than a table). The consumer knows more about the context it's using the `SortDirections` in either way, and if it's one where there are multiple columns managed by one consumer than it can always smuggle that context in via the callback (i.e. `sortFunction={(direction) => this.sort(relevant_col, direction)}` passed to each `SortDirections` on render).

`sortFunction` sounds like it performs the sorting, but it shouldn't. Most likely it's just a state update. Maybe `onDirectionClick`? Conveys that it's mostly an `onClick`, but that it has some relation to the direction concept too. Alternatively, maybe just make the consumer provide two click callbacks, `onClickDescending` and `onClickAscending`? Again, the consumer will hold all the information and can smuggle whatever it wants in via the callbacks, so we might as well separate brains/behaviour and presentation as much as we can.

Reply:
Unless part of the idea here is specifically to move behaviour/smarts down in to this component? When I asked for links to the previous discussions, that's what I was trying to figure out. There's some benefits if that's the plan (enforced consistent behaviour between uses of this component), but IMO also downsides (lose the clean smart vs presentational component design, likely going to mean leaks/coupling between this component and consumers).